src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordConstructor.cs

```diff
@@ -148,16 +165,15 @@ public IEnumerable<FieldSymbol> GetBackingFields()
                     result.Add(parameter, new SynthesizedFieldSymbol(containingType, parameter.Type, name));
                 }
 
-                Interlocked.CompareExchange(ref _capturedParameters, result, null);
+                captured.Free();
+                return result;
             }
 
-            captured.Free();
-
-            return _capturedParameters;
+            throw ExceptionUtilities.Unreachable();
```

Comment:
>throw ExceptionUtilities.Unreachable(); [](http://example.com/codeflow?start=12&length=39)

This doesn't seem to be needed. Consider also lifting code out of the `else` block above - it's not needed since the `if` returns.